### DISCUSSION TITLE
How can I generate a complete CHANGELOG for an existing repo that has a brand new semantic release setup

### DISCUSSION THREAD

**Discussion by zackarydev** [semantic-release/semantic-release#2871] [Q&A] [answered]
Title is self explanatory IMO. Ideally I'd like to run this locally but OK to run in CI if its the only option.

**Comment by travi** [comment, marked as answer]
semantic-release only supports release notes as part of a release. You would need to look into using conventional-changelog directly for that initial generation, which is what we use for the additions per release. 

However, our recommendation is to avoid using a changelog _file_ if you can. If you are just looking to add one now to an existing project, you could consider continuing without that file and instead leveraging, for example, GitHub releases. That allows you to avoid the added complexity of committing changes with each release.
